dockstore-client/src/main/java/io/dockstore/client/cli/nested/ToolClient.java

```diff
@@ -210,28 +211,53 @@ protected void handleSearch(String pattern) {
     }
 
     protected void handlePublishUnpublish(String entryPath, String newName, boolean unpublishRequest) {
+        DockstoreTool existingTool = null;
+        boolean isPublished = false;
+
+        // Cannot be making an unpublish request with a specified new name
+        assert (!(unpublishRequest && newName != null));
+
+        try {
+            existingTool = containersApi.getContainerByToolPath(entryPath, null);
+            isPublished = existingTool.isIsPublished();
+        } catch (ApiException ex) {
+            exceptionMessage(ex, "Unable to " + (unpublishRequest ? "unpublish " : "publish ") + entryPath, Client.API_ERROR);
+        }
+
+        if (existingTool == null) {
```

Comment:
I don't think this if test is necessary. If line 221 fails to find a tool, it will throw an exception, and line 224 will exit. And if it didn't throw an exception, you'd get an NPE on line 222.

Reply:
I see what you're saying, thanks for the catch! Removing the unnecessary lines.